unittests/baseplatformtests.py

```diff
@@ -88,11 +88,11 @@ def setUpClass(cls) -> None:
 
         # Misc stuff
         if cls.backend is Backend.ninja:
-            cls.no_rebuild_stdout = ['ninja: no work to do.', 'samu: nothing to do']
+            cls.no_rebuild_stdout = frozenset({'ninja: no work to do.', 'samu: nothing to do'})
```

Comment:
Is there a genuine worry that this is going to be modified? Invoking frozenset() seems interesting in a commit claiming to be speeding things up. ;)

Reply:
In my benchmarks using frozenset was slightly faster than a normal set for doing the lookups, and was only slightly slower for creating than a regular set. Since the frozenset only gets created once per class that seems like a pretty good tradeoff. If you're still concerned about that, this actually only gets used in `allplatformtests.py`, so we could move it to that class to speed things up even more.

Reply:
I guess I just wasn't sure why a set isn't good enough, that we need a frozenset. :P

You say a frozenset is slightly faster but I was under the impression that they both use the same CPython implementation and that frozensets are just sets with the mutability methods removed and `__hash__` added so that you can use `mydict[frozenset({'one'})]`. So I would be surprised if there was any performance difference that wasn't actually environment noise e.g. system load at the time the benchmark was done.

The real difference between the two is that sets have syntax for construction, whereas frozensets always incur a name lookup and function call.

Reply:
Maybe I've just been doing too much C++ and Rust recently, and would rather have immutable than mutable types by default :D